### ISSUE TITLE
Documenting path parameters

### ISSUE THREAD

**Issue by YnkDK** [jmcarp/flask-apispec#137]
How can I document path parameters? I expected the following to work:

````python
from apispec import APISpec
from apispec.ext.marshmallow import MarshmallowPlugin
from flask import Flask, Blueprint
from flask_apispec import FlaskApiSpec, MethodResource, doc

app = Flask(__name__)
api_blueprint = Blueprint('api', __name__, url_prefix='/api/')

app.config.update({
    'APISPEC_SPEC': APISpec(
        'Pets pets pets',
        '0.0.1',
        '2.0',
        [MarshmallowPlugin()]
    )
})


class PetResource(MethodResource):
    @doc(
        parameters=[
            {
                'name': 'pet_id',
                'in': 'path',
                'description': 'The ID of the pet',
                'required': True,
                'type': 'string',
                'example': '0c3ca635-2d58-4548-ad59-027fe4983e2c'
            }
        ]
    )
    def get(self, **kwargs):
        return None


pet_view = PetResource.as_view(PetResource.__name__)
api_blueprint.add_url_rule('/pet/<string:pet_id>', view_func=pet_view)

app.register_blueprint(api_blueprint)

docs = FlaskApiSpec(app)
docs.register(PetResource, endpoint=PetResource.__name__, blueprint=api_blueprint.name)

app.run(debug=True)
````

but this produces the following swagger:
````json
{
  "definitions": {}, 
  "info": {
    "title": "Pets pets pets", 
    "version": "0.0.1"
  }, 
  "parameters": {}, 
  "paths": {
    "/api/pet/{pet_id}": {
      "get": {
        "parameters": [
          {
            "in": "path", 
            "name": "pet_id", 
            "required": true, 
            "type": "string"
          }
        ], 
        "responses": {}
      }
    }
  }, 
  "responses": {}, 
  "securityDefinitions": {}, 
  "swagger": "2.0", 
  "tags": []
}
````
As seen I would like to be able to provide a description and an example value.

**Comment by AbdealiJK** [comment]
Doing
```
paramx={'pet_id': 
            {
                'name': 'pet_id',
                'in': 'path',
                'description': 'The ID of the pet',
                'required': True,
                'type': 'string',
                'example': '0c3ca635-2d58-4548-ad59-027fe4983e2c'
            }
        }
```
worked for me

**Comment by eberhab** [comment]
This is really helpful. Where is this documented? I am particularly looking for a way to change other parameter properties like "required", or "type". Also, for me "example" did not work, after some experimenting, I found that I need to name it "default". It would be great if all available options would be listed somewhere, but it does not seem to be in the official apispec docs. Where did you find this?

**Comment by edrandall** [comment]
@eberhab  Did you ever find anything?   Also looking for same.

**Comment by eberhab** [comment]
No, never found a comprehensive version dependent overview. I figured out a few things by guessing. What exactly are you looking for? Can try to share a few examples if it is something that we managed to solve.

**Comment by edrandall** [comment]
My issue stems from an older project on Python 3.7 using end-of-life dependencies and trying to incrementally upgrade (python, flask, sqlalchemy, marshmallow, ...) without breaking the API.   

Studying the changelogs, I figured we should be able to at least upgrade from `flask-apispec==0.7.0` to `0.9.0` and  `apispec==0.39.0` to `3.3.2` without any major barriers.   But (making only these dependency version changes)  on start I immediately get an error:
```
TypeError: fields2parameters() got an unexpected keyword argument 'description'
```
stemming from calls to `flask-apispec.docs.register`.   The `@doc` annotations on the problem endpoints look like:
```
@doc(
    summary="text describing the endpoint",
    params={
        "id": {"description":"text describing the id parameter"}
    },
)
```

From the error message my guess is that the @doc annotation is somehow incorrect with the newer version, but without any document of what's allowed here I'm probably going to have to study source code.

**Comment by eberhab** [comment]
While I cannot answer your question specifically, maybe it helps to share what we worked out so far, using the following versions: `flask-apispec==0.11.4`, `apispec==6.0.1`, `marshmallow==3.18.0`, and `marshmallow-dataclass==8.3.0`.

For our endpoints, we use a combination of multiple decorators:

     from flask_apispec import use_kwargs, marshal_with, doc
     @use_kwargs(request_input.Schema(many=True))
     @doc(description=('Text describing the endpoint'), tags=["Endpoint category"])
     @marshal_with(request_result.Schema(many=True))
     @app.route("/url/of/endpoint", methods=["POST"])

The schemas are defined as follows (with a few random keys that we use):

     from dataclasses import field
     from marshmallow_dataclass import dataclass
     @dataclass
     class request_input:
         id: str = field(metadata={
             'description': 'text describing the id parameter',
             'example': 'example value',
             'validate'=validate.OneOf(["example value", "value2"]),
             'required': False,
             'is_nullable': False,
             'missing': "value2"
         })

The use of dataclasses also let's use validation methods and easily re-use the class definitions in our endpoints. Our custom datatypes we define via `marshmallow_dataclass.NewType`

Hope this helps! Let me know if you have any questions or need more details.
